src/replica/HttpDirectorIndexModule.cc

```diff
@@ -78,123 +78,114 @@ json HttpDirectorIndexModule::_buildDirectorIndex() {
+        string const msg =
+                "director table has not been properly configured in database '" + database.name + "'";
+        throw HttpError(__func__, msg);
+    }
+    string const primaryKeyColumn = table.directorTable.primaryKeyColumn();
+    string primaryKeyColumnType = string();
+    string const chunkIdColNameType = "INT";
+    string subChunkIdColNameType = string();
+    if (table.columns.empty()) {
+        string const msg =
+                "no schema found for director table '" + table.name + "' of database '" + database.name + "'";
+        throw HttpError(__func__, msg);
+    }
+    for (auto&& column : table.columns) {
+        if (column.name == primaryKeyColumn) {
+            primaryKeyColumnType = column.type;
+        } else if (column.name == lsst::qserv::SUB_CHUNK_COLUMN) {
+            subChunkIdColNameType = column.type;
         }
     }
+    if (primaryKeyColumnType.empty() || subChunkIdColNameType.empty()) {
+        throw HttpError(__func__,
+                        "column definitions for the director key or sub-chunk identifier"
+                        " columns are missing in the director table schema for table '" +
+                                table.name + "' of database '" + database.name + "'");
+    }
 
     // Build/rebuild the index(es).
-
     bool const noTransactions = false;
     bool const allWorkers = true;
     TransactionId const noTransactionId = 0;
-    json extError = json::object();
     bool failed = false;
-    for (auto&& tableName : directorTables) {
-        // The entry will have the empty object for the tables if no problems will be
-        // encountered during the construction of the index.
-        extError[tableName] = json::object();
-
-        // Manage the new connection via the RAII-style handler to ensure the transaction
-        // is automatically rolled-back in case of exceptions.
-        ConnectionHandler const h(qservMasterDbConnection("qservMeta"));
-        QueryGenerator const g(h.conn);
-        auto const indexTableName = directorIndexTableName(database.name, tableName);
-
-        // (Re-)create the index table. Note that the table creation statement (the way it's
-        // written below) would fail if the table already exists. Hence, dropping it in
-        // the 'rebuild' mode should be explicitly requested by a client to avoid the problem.
-        vector<string> queries;
-        if (rebuild) {
-            bool const ifExists = true;
-            queries.push_back(g.dropTable(indexTableName, ifExists));
+
+    // The object will stay empty if no problems will be encountered during the construction
+    // of the index.
+    json extError = json::object();
+
+    // Manage the new connection via the RAII-style handler to ensure the transaction
+    // is automatically rolled-back in case of exceptions.
+    ConnectionHandler const h(qservMasterDbConnection("qservMeta"));
+    QueryGenerator const g(h.conn);
```

Comment:
I'm going to argue against the single letter variable name. There's no way to search for 'h' or 'g` reliably. 'conH' and 'qGen' would be significant improvements.

Reply:
Good point. Will fix it.